recipes/ado-mode

```diff
@@ -0,0 +1,5 @@
+(ado-mode
+ :fetcher github
+ :repo "louabill/ado-mode"
+ :files ("lisp/*.el" "scripts" "syntax_testers" "templates"
+		 (:exclude ".DS_Store" "*/.DS_Store" "*/Scratch")))
```

Comment:
If you have any .DS_Store files in your GitHub repo they should be removed and added to your `.gitignore`, not part of your recipe.

I'm not seeing any files called `*/Scratch` for you to exclude, so that part might be worth excluding as well -- generally the simpler the recipe, the more flexible it can be for you without having to open followup PRs.

```suggestion
 :files ("lisp/*.el" "scripts" "syntax_testers" "templates"))
```

But while I'm here, are the files in `syntax_testers` part of this package, or are they for testing?  We try to avoid packing up test files.

Reply:
Hmm... putting in the `.DS_Store` and `*/Scratch` were in my `.gitignore` already. I think I was being stupid and duplicating items just to protect against absent-mindedness. I'll fix the recipe.

The syntax testers were always a part of the package in the past, simply so others could see what worked and what did not. Stata is not a regular language, and hence is full of irregular constructions, some of which are impossible to fontify. The syntax testers can be useful for people running into these. Still, I get that they would not be used by 99% of the people downloading the package. I'll move them into another git repo for people to get separately.

Thanks for the comments and info.

Reply:
How do I run `package-lint` on the files in the package without getting the errors on the non-main files that variables do not have the proper prefix? For example, how do I avoid errors like `84:0: error: "ado-start-cmd-no-prefix-regexp" doesn't start with package's prefix "ado-cons".`? I feel rather stupid, because I don't recall the errors when I linted earlier, but then maybe I just ignored the errors.

Reply:
I would say don't feel that way -- even expert elisp programmers run into these gotchas, partly I think because Emacs keeps state so it's easy for run to run into "well it works on my machine" discrepancies.

The way you avoid that error is by setting the `package-lint-main-file` variable -- per its docstring:
```
  This variable is safe as a [file local variable](https://www.emacswiki.org/emacs/FileLocalVariables) if its value
  satisfies the predicate ‘stringp’.
```
so you can set it inside the files.  When I test out packages I have a [script](https://github.com/riscy/melpazoid) that identifies the main file and sets it automatically.

Reply:
For completeness:

The syntax testers were really for those who wanted to see what effect their customizations would have with `ado-mode`. I've split them out and put them in another repo.

I also set the `package-lint-main-file` in each of the non-main `.el` files to shut off the errors. Thanks for pointing me in the right direction.